hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokeninfo/TokenInfoCall.java

```diff
@@ -0,0 +1,80 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.tokeninfo;
+
+import static com.hedera.hapi.node.base.ResponseCodeEnum.SUCCESS;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.HederaSystemContract.FullResult.revertResult;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.HederaSystemContract.FullResult.successResult;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.TokenTupleUtils.tokenInfoTupleFor;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.tokeninfo.TokenInfoTranslator.TOKEN_INFO;
+import static com.swirlds.common.utility.CommonUtils.hex;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.ResponseCodeEnum;
+import com.hedera.hapi.node.state.token.Token;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.HederaSystemContract.FullResult;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.AbstractNonRevertibleTokenViewCall;
+import com.hedera.node.app.service.contract.impl.hevm.HederaWorldUpdater;
+import com.hedera.node.config.data.LedgerConfig;
+import com.swirlds.config.api.Configuration;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+public class TokenInfoCall extends AbstractNonRevertibleTokenViewCall {
+    private final Configuration configuration;
+    private final boolean isStaticCall;
+
+    public TokenInfoCall(
+            @NonNull final HederaWorldUpdater.Enhancement enhancement,
+            final boolean isStaticCall,
+            @Nullable final Token token,
+            @NonNull final Configuration configuration) {
+        super(enhancement, token);
+        this.configuration = configuration;
+        this.isStaticCall = isStaticCall;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected @NonNull FullResult resultOfViewingToken(@NonNull final Token token) {
+        requireNonNull(token);
+        // TODO - gas calculation
+
+        return fullResultsFor(SUCCESS, 0L, token);
+    }
+
+    @Override
+    protected @NonNull FullResult viewCallResultWith(
+            @NonNull final ResponseCodeEnum status, final long gasRequirement) {
+        return fullResultsFor(status, gasRequirement, Token.DEFAULT);
+    }
+
+    private @NonNull FullResult fullResultsFor(
+            @NonNull final ResponseCodeEnum status, final long gasRequirement, @NonNull final Token token) {
```

Comment:
This is a bit tricky - first time I've spotted it even though it showed up before I think - but this `token` argument is `@NonNull` even though the `token` given to the constructor is `@Nullable` because `AbstractNonRevertibleTokenViewCall` only calls `viewCallResultWith` if the `token != null` and that calls this method.

There's no real way to annotate this, you just have to follow the code around to see it (especially considering the field `token` all the way up at `AbstractNonRevertibleTokenViewCall` is `@Nullable`). Still, a `requireNonNull` might be helpful sometime.